picker.go

```diff
@@ -0,0 +1,56 @@
+package grpc
+
+import (
+	"time"
+)
+
+// Picker picks a Conn for RPC requests.
+// This is EXPERIMENTAL and Please do not implement your own Picker for now.
+type Picker interface {
+	// Pick returns the Conn to use for the upcoming RPC. It may return different
+	// Conn's up to the implementation.
+	Pick() (*Conn, error)
```

Comment:
Does the Picker need to know anything else about the request in order to select the Conn, e.g. query cost or perhaps a sticky-backend scheduling token?

What happens if there are no available Conns (e.g. due to all of them redialing)?  Should it return some particular error immediately, or block until one becomes available?